### PR TITLE
Add new file mode, add usage and library docs

### PR DESCRIPTION
**Pull request by kanaka**
* Add support for file deps data from files (JSON).
* Refactor library functions to make them more externally usable
* Add guide/examples to README with 9 CLI example
* Document library functions with Clojure usage examples
* Add comma separation for paths (in addition to colon).
* Add map format for alternate/or deps

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,355 @@
...
+weak dependencies (order only).
+
+Equivalent implmentations of resolve-deps are provided in both
+ClojureScript and Python. The resolution algorithm is very
+inefficient when resolving alternate resolution paths.
+
+## Guide / Examples
+
+In the following examples, you can use `./resolve-deps.py`
+interchangeably with `./resole-deps`.
```
Comment (jonsmock) [inline README.md]: ./resolve-deps

**Review by jonsmock — APPROVED** [review]

**Comment by jonsmock** [comment]
Awesome work!  Examples are excellent.

File: README.md (hunk 1 above)
Comment (kanaka) [reply README.md]: Fixed.

**Comment by kanaka** [comment]
@jonsmock Thanks! I've added another commit to bump to 0.1.0 and merged. When you get a chance (no hurry) can you npm publish this new version?

**Pull request merged**